File: Examples/VisualBasic/ModifyingAndConvertingImages/DICOM/BinarizationWithFixedThresholdOnDICOMImage.vb

Imports Aspose.Imaging.FileFormats.Dicom
Imports Aspose.Imaging.FileFormats.Tiff
Imports Aspose.Imaging.ImageFilters.FilterOptions
Imports Aspose.Imaging.ImageOptions

'
'This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Imaging for .NET API reference 
'when the project is build. Please check https:// Docs.nuget.org/consume/nuget-faq for more information. 
'If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from http://www.aspose.com/downloads, 
'Install it and then add its reference to this project. For any issues, questions or suggestions 
'Please feel free to contact us using http://www.aspose.com/community/forums/default.aspx
'


Namespace Aspose.Imaging.Examples.VisualBasic.ModifyingAndConvertingImages.DICOM
    Class BinarizationWithFixedThresholdOnDICOMImage
        Public Shared Sub Run()
            ' ExStart:BinarizationWithFixedThresholdOnDICOMImage
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_DICOM()

            ' Load an existing image.
            Using image As New DicomImage(dataDir & Convert.ToString("image.dcm"))
                ' Binarize image with predefined fixed threshold.
                image.BinarizeFixed(100)

                ' Save the resultant image.
                image.Save(dataDir & Convert.ToString("BinarizationWithFixedThresholdOnDICOMImage_out.bmp"), New BmpOptions())
            End Using
            ' ExEnd:BinarizationWithFixedThresholdOnDICOMImage
        End Sub
    End Class
End Namespace

File: Examples/CSharp/ModifyingAndConvertingImages/DICOM/BinarizationWithFixedThresholdOnDICOMImage.cs
using Aspose.Imaging.FileFormats.Dicom;
using Aspose.Imaging.ImageOptions;
using System;
using System.IO;

/*
This project uses the Automatic Package Restore feature of NuGet to resolve the Aspose.Imaging for .NET API reference when the project is built. Please see https://learn.microsoft.com/en-us/nuget/resources/nuget-faq for more information.
If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from https://releases.aspose.com/, install it, and then add its reference to this project. For any issues, questions, or suggestions, please feel free to contact us via https://forum.aspose.com/.
*/

namespace Aspose.Imaging.Examples.CSharp.ModifyingAndConvertingImages.DICOM
{
    class BinarizationWithFixedThresholdOnDICOMImage
    {
        public static void Run()
        {
            //ExStart:BinarizationWithFixedThresholdOnDICOMImage
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_DICOM();

            Console.WriteLine("Running example BinarizationWithFixedThresholdOnDICOMImage");
            using (var fileStream = new FileStream(dataDir + "file.dcm", FileMode.Open, FileAccess.Read))
            using (DicomImage image = new DicomImage(fileStream))
            {
                // Binarize the image with a predefined fixed threshold and save the resulting image.
                image.BinarizeFixed(100);
                image.Save(dataDir + "BinarizationWithFixedThresholdOnDICOMImage_out.bmp", new BmpOptions());
            }

            Console.WriteLine("Finished example BinarizationWithFixedThresholdOnDICOMImage");
            //ExEnd:BinarizationWithFixedThresholdOnDICOMImage
        }
    }
}